lib/data/models/approval_request.dart

```diff
@@ -30,28 +30,28 @@ class ApprovalRequest with _$ApprovalRequest {
 
   const factory ApprovalRequest.signData({
     required String origin,
-    required String publicKey,
+    required PublicKey publicKey,
     required String data,
     required Completer<String> completer,
   }) = _SignData;
 
   const factory ApprovalRequest.encryptData({
     required String origin,
-    required String publicKey,
+    required PublicKey publicKey,
     required String data,
     required Completer<String> completer,
   }) = _EncryptData;
 
   const factory ApprovalRequest.decryptData({
     required String origin,
-    required String publicKey,
-    required String sourcePublicKey,
+    required PublicKey publicKey,
+    required PublicKey sourcePublicKey,
     required Completer<String> completer,
   }) = _DecryptData;
 
   const factory ApprovalRequest.callContractMethod({
     required String origin,
-    required String publicKey,
+    required PublicKey publicKey,
     required String recipient,
```

Comment:
I dont remember, but `recipient` is address or public key, check it in old app pls

Reply:
And `origin` too

Reply:
Origin is URL.authority that is formatted from the [userInfo], [host] and [port]. So, it is just a string 🤷

Reply:
Changed sender and recipient to dataclass